.github/workflows/release.yml

```diff
@@ -1,25 +1,96 @@
-name: release
+name: Create Release
 
 on:
   push:
-    # Sequence of patterns matched against refs/tags
-    tags:
-    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
-
+    branches:
+      - main
+    paths:
+      - 'CHANGELOG/*.md'
+  
 permissions:
-  contents: write # Allow to create a release.
+  contents: write # Allow to push a tag.
 
 jobs:
-  build:
+  push_release_tags:
+    runs-on: ubuntu-latest
+    outputs:
+      release_tag: ${{ steps.release-version.outputs.release_version }}
+    steps:
+      - name: Checkout code 
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # tag=v3.5.3
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@920e7b9ae1d45913fc81f86c956fee89c77d2e5e # tag=v37.5.0
+      - name: Get release version
+        id: release-version
+        run: |
+            if [[ ${{ steps.changed-files.outputs.all_changed_files_count }} != 1 ]]; then
```

Comment:
@CecileRobertMichon Do you think it would make sense to move large parts of the scripting into a shell script so it's easier to test locally / maintain?

Reply:
I could although I don't think you'd really want to run any of these scripting steps as standalone since they pretty much all use environment variables (e.g. `"$GITHUB_ACTOR"`) that are only available if running in the context of a github workflow or are outputs from the previous steps/job.

The way I've been testing this on my fork is by changing the references to "main" branch to my feature branch and then pushing new release notes for a fake tag version in `CHANGELOG/` to my branch. This triggers a tag and a release to be created though so it requires you to do some cleanup if you don't want to pollute your fork with test tags.